react/src/hooks/useMemoWithPrevious.tsx

```diff
@@ -0,0 +1,36 @@
+import { DependencyList, useEffect, useMemo, useRef, useState } from 'react';
+
+export const useMemoWithPrevious = <T,>(
+  factory: () => T,
+  { deps, initialPrev }: { deps: DependencyList; initialPrev?: T },
+) => {
+  const prevRef = useRef(initialPrev);
+  const [prevResetKey, setPrevResetKey] = useState({});
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const current = useMemo(factory, deps);
+  const memoizedPrev = useMemo(() => {
+    return prevRef.current;
+    // Only update when the reset key changes and deps change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [...deps, prevResetKey]);
+
+  useEffect(() => {
+    prevRef.current = current;
+    // Only update when deps change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [deps]);
```

Comment:
According to the comment, I think it should be the same as above.
```suggestion
  }, [...deps]);
```

Reply:
Oh! I found a big my mistake. ;)